engine/include/cubos/engine/imgui/data_inspector.hpp

```diff
@@ -0,0 +1,66 @@
+/// @file
+/// @brief Resource @ref cubos::engine::DataInspector.
+/// @ingroup imgui-plugin
+
+#pragma once
+
+#include <cubos/core/memory/any_value.hpp>
+#include <cubos/core/reflection/type.hpp>
+
+namespace cubos::engine
+{
+    /// @brief Resource which allows the user to inspect or modify any reflectable value in the UI.
+    /// @see Take a look at the @ref examples-engine-imgui example for a demonstration of this resource.
+    /// @ingroup imgui-plugin
+    class DataInspector final
+    {
+    public:
+        /// @brief Inspect a reflective object.
+        ///
+        /// @note This should always be called inside a `ImGui::BeginTable(..., 2)` and `ImGui::EndTable()`.
+        /// @param name The name of the object.
+        /// @param type The reflection type of the object.
+        /// @param value A pointer to the object's value.
+        /// @param readOnly A flag that indicates wether this field can be modified.
+        /// @return True if object was modified.
+        bool inspect(const std::string& name, const core::reflection::Type& type, void* value, bool readOnly = true);
```

Comment:
I would keep inspect something internal and here have only `show` and `edit`, show receiving `const void*`. This way the user doesn't have to use const_cast.

Reply:
Would also be cool to add template methods with signatures `show(const std::string& name, const T& value);` which call reflect for us.